### DISCUSSION TITLE
UK Postcodes Don’t Work

### DISCUSSION THREAD

**Discussion by mikekenyon99** [osmandapp/OsmAnd#19503] [Maps Related]
I have tried to use OSMand for navigation for postcodes and I must be missing something.  For example I put a Sheffield City Centre postcode and it doesn’t find it - it finds some postcodes that I type in though.  It’s like a totally random pot luck.  

Give it a S1 postcode and it pulls up some, put S2 and it says nothing within 50 miles.  For reference I live 10 miles from the centre of Sheffield.   I have downloaded the maps so I can work offline. 

There clearly has been a lot of work put into this but it simply fails to find postcodes, I must be missing something.  

Any thoughts?

**Comment by tommycrock** [comment]
Hi @mikekenyon99 

The workaround is to download the postcodes as POIs from [Harry van der Wolf's github repository](https://github.com/hvdwolf/OsmAnd-UKpostcodes/releases) as advertised on the [OsmAnd google group](https://groups.google.com/g/osmand/c/ITkXvkPYb8I/m/edt7VtOPAAAJ). You'll probably have to find the downloaded obf and 'open with'... on your phone, and select OsmAnd to get it to install successfully. Note that when using these you just have to search generally, not using the structured postcode search.

The reason it's like this is that osmand is based on the data in [OpenStreetMap](https://www.openstreetmap.org/about) which only contains some UK postcodes because address data is not open source in the UK and the UK community agreed not to import postcode centroids like these.

**Reply by mikekenyon99** [reply]
Thank you very much for explaining that :-) I knew I was missing something.

**Reply by djSupport** [reply]
I just found this whilst looking into the same problem but if I go to the open street map website and search by postcode it finds me quickly, why is this?

**Reply by tommycrock** [reply]
Nominatim (the search engine on the osm website) imports UK postcodes in addition to osm data. https://github.com/osm-search/gb-postcode-data

**Reply by djSupport** [reply]
Can osmand not use this to process postcodes? The most common form of navigation if using android auto?
